numpy/lib/npyio.py

```diff
@@ -733,10 +733,15 @@ def _getconv(dtype):
     """ Find the correct dtype converter. Adapted from matplotlib """
 
     def floatconv(x):
-        x.lower()
-        if '0x' in x:
-            return float.fromhex(x)
-        return float(x)
+        try:
+            return float(x)  # The fastest path.
+        except ValueError as exc:
+            try:
+                if '0x' in x:  # Don't accidentally convert "a" ("0xa") to 10.
```

Comment:
I think it would be clearer with this if outside the try

Reply:
sure